language-service/src/services/yamlCompletion.ts

```diff
@@ -98,518 +98,523 @@ export class YAMLCompletion {
-			//console.log('schema: ' + JSON.stringify(schema));
-			if (schema) {
-				this.getValueCompletions(schema, currentDoc, node, offset, document, collector, types);
-			} 
-			if (this.contributions.length > 0) {
-				this.getContributedValueCompletions(currentDoc, node, offset, document, collector, collectionPromises);
-			}
-			if (this.customTags.length > 0) {
-				this.getCustomTagValueCompletions(collector);
-			}
-
-			return this.promise.all(collectionPromises).then(() => {
-				return result;
-			});
-		});
-	}
-
-	private arrayIsEmptyOrContainsKey(stringArray: string[], key: string): boolean {
-		if (!stringArray || !stringArray.length) {
-			return true;
-		}
-
-		return !!stringArray.find(arrayEntry => arrayEntry === key);
-	}
-
-	private getPropertyCompletions(schema: SchemaService.ResolvedSchema, doc, node: Parser.ASTNode, addValue: boolean, collector: CompletionsCollector, separatorAfter: string): void {
-		const matchingSchemas = doc.getMatchingSchemas(schema.schema);
-		matchingSchemas.forEach((s) => {
-			if (s.node === node && !s.inverted) {
-				const schemaProperties = s.schema.properties;
-				if (schemaProperties) {
-					Object.keys(schemaProperties).forEach((key: string) => {
-						//check for more than one propery because the placeholder will always be in the list
-						if (s.node.properties.length > 1 || this.arrayIsEmptyOrContainsKey(s.schema.firstProperty, key)) {
-							const propertySchema = schemaProperties[key];
-							if (!propertySchema.deprecationMessage && !propertySchema["doNotSuggest"]) {
-								collector.add({
-									kind: CompletionItemKind.Property,
-									label: key,
-									insertText: this.getInsertTextForProperty(key, propertySchema, addValue, separatorAfter),
-									insertTextFormat: InsertTextFormat.Snippet,
-									documentation: propertySchema.description || ''
-								});
-							}
-						}
-					});
-				}
-			}
-		});
-	}
-
-	private getValueCompletions(schema: SchemaService.ResolvedSchema, doc, node: Parser.ASTNode, offset: number, document: TextDocument, collector: CompletionsCollector, types: { [type: string]: boolean } ): void {
-		
-
-		let offsetForSeparator = offset;
-		let parentKey: string = null;
-		
-		if (node && (node.type === 'string' || node.type === 'number' || node.type === 'boolean')) {
-			offsetForSeparator = node.end;
-			node = node.parent;
-		}
-
-		if(node && node.type === 'null'){
-			let nodeParent = node.parent;
-			
+        } else {
+            //console.log('else');
+            let overwriteStart = offset - currentWord.length;
+            if (overwriteStart > 0 && document.getText()[overwriteStart - 1] === '"') {
+                overwriteStart--;
+            }
+            overwriteRange = Range.create(document.positionAt(overwriteStart), position);
+        }
+
+        //console.log('overwriteRange: ' + JSON.stringify(overwriteRange));
+
+        let proposed: { [key: string]: CompletionItem } = {};
```

Comment:
What happened to all of the spacing? Was it wrong before?

Reply:
yes, it was all tabs... Hope we can agree that we want spaces?